opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java

```diff
@@ -92,9 +89,27 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
         }
     }
 
+    private SpanContext parentSpanContext(ContainerRequestContext requestContext) {
+        SpanContext parentSpanContext = null;
+
+        Span activeSpan = tracer.activeSpan();
+        if (activeSpan != null) {
+            parentSpanContext = activeSpan.context();
+        } else {
+            SpanContext extractedSpanContext = tracer.extract(
+                Format.Builtin.HTTP_HEADERS,
+                new ServerHeadersExtractTextMap(requestContext.getHeaders())
+            );
+            if (extractedSpanContext != null) {
```

Comment:
nit: This if looks redundant.

This method looks could be a lot simpler without SESE. Just return if active is not `null`. The default return can be direct `return tracer.extract()`

Reply:
you're right, it's because I extracted this method and didn't review it later ;)